### PR TITLE
feat: Add `/_fake/update_action_attempt`

### PR DESCRIPTION
**Pull request by andrii-balitskyi**
- **Add /_fake/update_action_attempt, update action_attempt zod schema**
- **Add test**

### CODE REVIEW

File: src/pages/api/_fake/update_action_attempt.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  if (action_attempt == null) {
+    throw new NotFoundException({
+      type: "action_attempt_not_found",
+      message: "Action attempt not found",
+    })
+  }
+
+  req.db.updateActionAttempt({
+    action_attempt_id,
+    ...(action_attempt_update_payload as Partial<ActionAttempt>),
```
Comment (razor-x) [inline src/pages/api/_fake/update_action_attempt.ts:42]: Why do we need the type assertion here?

**Review by razor-x — APPROVED** [review]

File: src/pages/api/_fake/update_action_attempt.ts (hunk 1 above)
Comment (andrii-balitskyi) [reply src/pages/api/_fake/update_action_attempt.ts:42]: I get this type error.
image

We can probably fix this with something like: 
```
  req.db.updateActionAttempt({
    action_attempt_id,
    ...(action_attempt_update_payload.status === "error"
      ? {
          status: "error",
          error: action_attempt_update_payload.error,
          action_type: action_attempt_update_payload.action_type,
          result: null,
        }
      : action_attempt_update_payload.status === "success"
        ? {
            status: "success",
            error: null,
            action_type: action_attempt_update_payload.action_type,
            result: action_attempt_update_payload.result,
          }
        : action_attempt_update_payload.status === "pending"
          ? {
              status: "pending",
              error: null,
              action_type: action_attempt_update_payload.action_type,
              result: null,
            }
          : {}),
  })
  ```

File: src/pages/api/_fake/update_action_attempt.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+import { withRouteSpec } from "lib/middleware/with-route-spec.ts"
+import type { ActionAttempt } from "lib/zod/action_attempt.ts"
+
+export default withRouteSpec({
+  auth: "none",
+  methods: ["PATCH", "POST"],
+  jsonBody: z.object({
+    action_attempt_id: z.string(),
+    status: z.enum(["success", "error", "pending"]).optional(),
+    action_type: z.string().optional(),
```
Comment (razor-x) [inline src/pages/api/_fake/update_action_attempt.ts:13]: We shouldn't allow updating the action type

File: src/pages/api/_fake/update_action_attempt.ts (hunk 1 above)
Comment (andrii-balitskyi) [reply src/pages/api/_fake/update_action_attempt.ts]: https://github.com/seamapi/fake-seam-connect/pull/207/commits/87d7d8fc4e67bc4e30d033564c9f3638210ec748

**Review by razor-x — APPROVED** [review]
👌🏻

**Pull request merged**